bin/dfx-software-nns-dapp-install

```diff
@@ -0,0 +1,82 @@
+# Source the output file ----------------------------------------------------------
+source "$(clap.build)"
+
+ND_WASM_FILENAME="nns-dapp.wasm"
+ND_WASM_URL="$(dfx-software-nns-dapp-ci-wasm-url --release "${DFX_ND_RELEASE}")"
+curl --retry 5 --fail -sSL "${ND_WASM_URL}" >"$ND_WASM_FILENAME"
+
+: There is no standardization for how args are constructed. We can use a pre-built set of arguments as long as all canister IDs, feature flags etc are exactly as in the upstream build. E.g. there is no standardized way of passing in the IDs of a list of canisters.
+case "$DFX_NETWORK" in
+ic | mainnet) ND_ARG_FILENAME=nns-dapp-arg-mainnet.bin ;;
+*) ND_ARG_FILENAME=nns-dapp-arg-local.bin ;;
+esac
+
+if false; then # The PR for publishing arguments hasn't been merged yet
+  ND_ARG_URL="${ND_WASM_URL%/*}/${ND_ARG_FILENAME}"
+  curl --retry 5 --fail -sSL "${ND_ARG_URL}" >"$ND_ARG_FILENAME"
+  cp "$ND_ARG_FILENAME" nns-dapp-arg.wasm
+else
+  echo "WARNING:  Using a toy config.  Please get the real arguments published and use those." >&2
+  cat <<-EOF >"$ND_ARG_FILENAME"
+	(opt record{
+	  args = vec {
+	    record{ 0="CYCLES_MINTING_CANISTER_ID"; 1="rkp4c-7iaaa-aaaaa-aaaca-cai" };
+	    record{ 0="DFX_NETWORK"; 1="local" };
+	    record{ 0="FEATURE_FLAGS"; 1="{\"ENABLE_CKBTC\":false,\"ENABLE_CKTESTBTC\":false,\"ENABLE_SIMULATE_MERGE_NEURONS\":false,\"ENABLE_SNS_AGGREGATOR\":true,\"ENABLE_SNS_VOTING\":true}" };
+	    record{ 0="FETCH_ROOT_KEY"; 1="true" };
+	    record{ 0="GOVERNANCE_CANISTER_ID"; 1="rrkah-fqaaa-aaaaa-aaaaq-cai" };
+	    record{ 0="HOST"; 1="http://localhost:8080" };
+	    record{ 0="IDENTITY_SERVICE_URL"; 1="http://qhbym-qaaaa-aaaaa-aaafq-cai.localhost:8080" };
+	    record{ 0="LEDGER_CANISTER_ID"; 1="ryjl3-tyaaa-aaaaa-aaaba-cai" };
+	    record{ 0="OWN_CANISTER_ID"; 1="qsgjb-riaaa-aaaaa-aaaga-cai" };
+	    record{ 0="OWN_CANISTER_URL"; 1="http://qsgjb-riaaa-aaaaa-aaaga-cai.localhost:8080" };
+	    record{ 0="ROBOTS"; 1="" };
+	    record{ 0="SNS_AGGREGATOR_URL"; 1="http://rwlgt-iiaaa-aaaaa-aaaaa-cai.localhost:8080" };
+	    record{ 0="TVL_CANISTER_ID"; 1="" };
+	    record{ 0="WASM_CANISTER_ID"; 1="qaa6y-5yaaa-aaaaa-aaafa-cai" };
+	  };
+	})
+EOF
+fi
+
+: "If the canister is not in dfx.json, it needs to be added."
+# TODO: Make a command that imports just the nns-dapp.
+dfx-nns-import
+
+: "If the canister was remote, it isn't any more."
+# Note: In previous versions of dfx, a __default remote entry did not prevent a canister from being created.  Now it does.
+for network in "${DFX_NETWORK}" __default; do
+  DFX_NETWORK="$network" jq 'del(.canisters["nns-dapp"].remote.id[env.DFX_NETWORK])' dfx.json | sponge dfx.json
+done
+
+: "Create canister, if needed"
+: "... it is easier to ask forgiveness than permission"
+dfx canister create nns-dapp --network "$DFX_NETWORK"
+: "Check that the canister seems to be present and healthy"
+if dfx canister info nns-dapp --network "$DFX_NETWORK"; then
+  echo "Canister seems to exist."
```

Comment:
Actually, `dfx canister create` makes the subnet agree that a canister exists at the given ID.  I believe it's fair to say that the canister exists if the replicas in the subnet say it does, even if it's empty.  Such a canister appears on the replica dashboard, for example.

Reply:
Honestly, if I see
```
Canister seems to exist.
```
on the output. I wouldn't even know if that's a good thing or a bad thing.
I might think the script is complaining that it can't install it because it's already there.
It might be less confusing to just not output anything here, or otherwise something more reassuring.

Reply:
Ok, I'll update the text to make it sound more confident.  I don't feel that confidence about canister deployments but we can smile at the passengers and tell them that all is OK.